### PR TITLE
add exercise 4

### PR DESCRIPTION
**Pull request by sroet**
adds exercise 4:
Binder link: https://mybinder.org/v2/gh/sroet/chemometrics/add_exercise_set_004?filepath=exercises

### CODE REVIEW

**Comment by sroet** [comment]
@andersle this is ready for review

File: exercises/04_Exercise_Set_4.ipynb [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+    "# Extract the two classes for plotting :\n",
+    "class1 = data[data['class'] == 'foo']\n",
+    "class2 = data[data['class'] == 'bar']\n",
+    "# We can plot the raw data as follows :\n",
+    "# Note: Here you could style the plot to your liking .\n",
+    "# Adding x- labels and y- labels is , for instance , a good idea.\n",
+    "plt.scatter(class1['x'], class1['y'], label='Class foo')\n",
+    "plt.scatter(class2['x'], class2['y'], label='Class bar')\n",
+    "plt.legend()\n",
+    "plt.show()"
```
Comment (andersle) [inline exercises/04_Exercise_Set_4.ipynb]: This line is not needed here.

```suggestion
```

File: exercises/04_Exercise_Set_4.ipynb [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.decomposition import PCA\n",
+    "plt.style.use('seaborn-talk')\n",
+    "# Load data:\n",
+    "data = pd.read_csv('Data/data_exercise4.txt', delim_whitespace =True)\n",
+    "variables = ['x','y'] # Insert the variables you will use here\n",
+    "X = data[variables]\n",
+    "# Uncomment the following line in order to scale the data:\n",
```
Comment (andersle) [inline exercises/04_Exercise_Set_4.ipynb]:
```suggestion
    "# Uncomment the following lines in order to scale the data:\n",
```

File: exercises/04_Exercise_Set_4.ipynb [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+    "# scaler = StandardScaler() # Initialise the scaler\n",
+    "# scaler.fit(X) # \"fit\" the scaller with the X data\n",
```
Comment (andersle) [inline exercises/04_Exercise_Set_4.ipynb]:
```suggestion
    "# scaler.fit(X) # \"fit\" the scaler with the X data\n",
```

**Comment by sroet** [comment]
@andersle ready for another round

**Pull request merged**